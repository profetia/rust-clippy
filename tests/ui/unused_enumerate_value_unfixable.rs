//@no-rustfix
#![warn(clippy::unused_enumerate_value)]

fn main() {
    struct Length(usize);

    impl IntoIterator for Length {
        type Item = usize;
        type IntoIter = std::iter::Once<usize>;

        fn into_iter(self) -> Self::IntoIter {
            std::iter::once(self.0)
        }
    }

    impl Length {
        fn len(&self) -> usize {
            self.0
        }
    }

    let length = Length(3);
    for (index, _) in length.into_iter().enumerate() {
        //~^ unused_enumerate_value
        todo!();
    }
}
