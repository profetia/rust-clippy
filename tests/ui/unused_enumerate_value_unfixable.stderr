error: you seem to use `.enumerate()` and immediately discard the value
  --> tests/ui/unused_enumerate_value_unfixable.rs:23:23
   |
LL |     for (index, _) in length.into_iter().enumerate() {
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider using `0..length.len()` instead
   = note: `-D clippy::unused-enumerate-value` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::unused_enumerate_value)]`

error: aborting due to 1 previous error

